package/pyproject.toml

```diff
@@ -0,0 +1,119 @@
+# PEP-518 https://peps.python.org/pep-0518/
+[build-system]
+# Minimum requirements for the build system to execute.
+requires = ["setuptools>=65.5.1"]  # PEP 518 specifications
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "kedro-viz"
+authors = [
+    {name = "Kedro"}
+]
+description = "Kedro-Viz helps visualise Kedro data and analytics pipelines"
+requires-python = ">=3.7"
+dependencies = [
+    "packaging~=23.0",
+    "kedro>=0.17.5",
+    "ipython>=7.0.0, <9.0",
+    "fastapi>=0.73.0, <0.96.0",
+    "fsspec[s3]>=2021.4, <2024.1",
+    "aiofiles==22.1.0",
+    "uvicorn[standard]~=0.22.0",
+    "watchgod~=0.8.2",
+    "plotly>=4.0",
+    "pandas>=1.3",
+    "sqlalchemy>=1.4, <3",
+    "strawberry-graphql>=0.192.0, <1.0",
+    "networkx>=2.5",
+    "orjson~=3.9",
+    "secure>=0.3.0",
+]
+keywords = [
+    "pipelines",
+    "machine learning",
+    "data pipelines",
+    "data science",
+    "data engineering",
+    "visualisation",
+]
+license = {text = "Apache Software License (Apache 2.0)"}
+dynamic = ["readme", "version"]
+
+[project.optional-dependencies]
+test = [
+    "kedro >=0.17.0",
+    "kedro-datasets[pandas.ParquetDataSet, pandas.CSVDataSet, pandas.ExcelDataSet, plotly.JSONDataSet]>=1.0, <1.7.1",
+    "kedro-telemetry>=0.1.1",  # for testing telemetry integration
+    "bandit~=1.7",
+    "behave~=1.2",
+    "black~=23.3",
+    "boto3~=1.26",
+    "flake8~=5.0",
+    "fastapi[all]>=0.73.0, <0.96.0",
+    "isort~=5.11",
+    "matplotlib~=3.5",
+    "mypy~=1.0",
+    "moto~=4.1.14",
+    "psutil==5.9.5",  # same as Kedro for now
+    "pylint~=2.17",
+    "pytest~=7.4",
+    "pytest-asyncio~=0.21",
+    "pytest-mock~=3.11",
+    "pytest-cov~=4.1",
+    "sqlalchemy-stubs~=0.4",
+    "strawberry-graphql[cli]>=0.99.0, <1.0",
+    "trufflehog~=2.2",
+    "types-aiofiles==0.1.3",
+    "types-cachetools==0.1.6",
+    "types-click==0.1.14",
+    "types-futures==0.1.3",
+    "types-Jinja2==2.11.2",
+    "types-orjson==0.1.0",
+    "types-pkg-resources==0.1.2",
+    "types-protobuf==0.1.10",
+    "types-PyYAML==0.1.5",
+    "types-requests==0.1.8",
+    "types-toml==0.1.1",
+    "types-ujson==0.1.0",
+    "typing_extensions~=4.7.0; python_version < '3.9'"
+]
+
+[tool.setuptools]
+zip-safe = false
+
+[tool.setuptools.packages.find]
+include = ["kedro_viz*"]
+exclude=["tests*", "features*"]
+
+[tool.setuptools.dynamic]
+readme = {file = "README.md", content-type = "text/markdown"}
```

Comment:
unable to read the README file as it is in a different directory. We need to consider following the directory structure of other kedro-plugins to be consistent since kedro-viz is also a plugin. 

@rashidakanchwala @noklam 

Our directory structure - `kedro-viz/package/kedro_viz`
Other plugins have (example) - `kedro-telemetry/kedro_telemetry` . 

Is it better we restructure our directory structure to `kedro-viz/kedro_viz` ?

Reply:
It's tricky in your case because there's a separation between the backend and the frontend 🤔 `../README.md` doesn't work?

Reply:
yes it throws an error saying toml cannot access any files outside the directory it is in

Reply:
you can use `setup.py` for reading the `README.md`, then setuptools will merge whatever is on `setup.py` with `pyproject.toml`. see what  `kedro-datasets` does for example